services/search/pkg/search/events.go

```diff
@@ -91,92 +95,74 @@ func (eh *eventHandler) trashItem(rid *provider.ResourceId) {
 	if err != nil {
 		eh.logger.Error().Err(err).Msg("failed to extract resource content")
 		return
 	}
 
 	r := engine.Resource{
-		ID:       storagespace.FormatResourceID(*statRes.Info.Id),
-		RootID:   storagespace.FormatResourceID(*ref.ResourceId),
-		Path:     ref.Path,
-		Type:     uint64(statRes.Info.Type),
+		ID: storagespace.FormatResourceID(*stat.Info.Id),
+		RootID: storagespace.FormatResourceID(provider.ResourceId{
+			StorageId: stat.Info.Id.StorageId,
+			OpaqueId:  stat.Info.Id.SpaceId,
+			SpaceId:   stat.Info.Id.SpaceId,
+		}),
+		Path:     utils.MakeRelativePath(path.Path),
+		Type:     uint64(stat.Info.Type),
 		Document: doc,
 	}
 
-	err = eh.engine.Upsert(r.ID, r)
-	if err != nil {
+	if err = eh.engine.Upsert(r.ID, r); err != nil {
 		eh.logger.Error().Err(err).Msg("error adding updating the resource in the index")
 	} else {
 		logDocCount(eh.engine, eh.logger)
 	}
 }
 
-func (eh *eventHandler) restoreItem(ref *provider.Reference, owner *user.User) {
-	if ref == nil || owner == nil {
+func (eh *eventHandler) restoreItem(ref *provider.Reference, uid *user.UserId) {
+	ctx, stat, path := eh.resInfo(uid, ref)
+	if ctx == nil || stat == nil || path == nil {
 		return
 	}
 
-	statRes, err := statResource(ref, owner, eh.gateway, eh.secret, eh.logger)
-	if err != nil {
-		eh.logger.Error().Err(err).Msg("failed to stat the changed resource")
-		return
+	if err := eh.engine.Restore(storagespace.FormatResourceID(*stat.Info.Id)); err != nil {
+		eh.logger.Error().Err(err).Msg("failed to restore the changed resource in the index")
 	}
+}
 
-	if statRes.Status.Code != rpc.Code_CODE_OK {
-		eh.logger.Error().Interface("statRes", statRes).Msg("failed to stat the changed resource")
+func (eh *eventHandler) moveItem(ref *provider.Reference, uid *user.UserId) {
+	ctx, stat, path := eh.resInfo(uid, ref)
+	if ctx == nil || stat == nil || path == nil {
 		return
 	}
 
-	err = eh.engine.Restore(storagespace.FormatResourceID(*statRes.Info.Id))
-	if err != nil {
-		eh.logger.Error().Err(err).Msg("failed to restore the changed resource in the index")
+	if err := eh.engine.Move(storagespace.FormatResourceID(*stat.Info.Id), path.Path); err != nil {
+		eh.logger.Error().Err(err).Msg("failed to move the changed resource in the index")
 	}
 }
 
-func (eh *eventHandler) moveItem(ref *provider.Reference, owner *user.User) {
-	if ref == nil || owner == nil {
-		return
-	}
-
-	statRes, err := statResource(ref, owner, eh.gateway, eh.secret, eh.logger)
+func (eh *eventHandler) resInfo(uid *user.UserId, ref *provider.Reference) (context.Context, *provider.StatResponse, *provider.GetPathResponse) {
+	ownerCtx, err := getAuthContext(&user.User{Id: uid}, eh.gateway, eh.secret, eh.logger)
 	if err != nil {
-		eh.logger.Error().Err(err).Msg("failed to stat the moved resource")
-		return
-	}
-	if statRes.Status.Code != rpc.Code_CODE_OK {
-		eh.logger.Error().Interface("statRes", statRes).Msg("failed to stat the moved resource")
-		return
+		return nil, nil, nil
```

Comment:
We should log the error before returning `nil, nil, nil` in all returns of `resInfo` func. Otherwise it will be hard debugging.

Reply:
error logging happens in getAuthContext, statResource and getPath;